naturalcrit/homebrewery
# Update Core.less and reset.less

(i've no idea how this got attached to "give dismisskeys a default", i apologize...)

## Description
This PR focuses on changes on `core.less` and `reset.less` to make styling easier across the project, particularly around `button` elements.  The PR is actually a cherry-pick from the last commit on #3868, a commit that I wish I could benefit from right now rather than waiting on that one.

Currently, if you use a `button` in HB it gets styled like this, without any additional input from you:
<img width="475" alt="image" src="https://github.com/user-attachments/assets/89a3eda7-3b38-4358-9df9-a37e8553b3b6">

All well and good when our site only has a few actual `button` elements, but a pain in the arse to unset all those properties if you want to use buttons in, say, a toolbar.  

This PR changes the default styling of buttons so that they basically lose their user-agent properties, but otherwise leaves them stripped bare.  The current green default button is then assigned the mixin `.colorButton()` and given the same parameter.  So when you actually *want* that style of button, you can ask for it.

Publish button, without using the mixin:
<img width="475" alt="image" src="https://github.com/user-attachments/assets/e15ff3fc-9a16-4f0e-bb9a-d8a41d812387">

(obviously the mixin will be used here, this is just for an example).


## QA Instructions, Screenshots, Recordings

Check that any place you expect a `button` to be it is still styled as you remember it.
